src/lib/server/map.c

```diff
@@ -164,11 +165,25 @@ int map_afrom_cp(TALLOC_CTX *ctx, map_t **out, map_t *parent, CONF_PAIR *cp,
 	 *	RHS might be an attribute reference.
 	 */
 	type = cf_pair_value_quote(cp);
+	p_rules = value_parse_rules_unquoted[type]; /* We're not searching for quotes */
+	if (type == T_DOUBLE_QUOTED_STRING || type == T_BACK_QUOTED_STRING) {
+		char *unescaped_value;
+		slen = fr_sbuff_out_aunescape_until(map, &unescaped_value,
+				&FR_SBUFF_IN(value, strlen(value)), SIZE_MAX, p_rules->terminals, p_rules->escapes);
+		if (slen < 0) {
+			marker_subject = value;
+			goto marker;
+		}
+		value = unescaped_value;
+		p_rules = NULL;
+	} else {
+		slen = strlen(value);
+	}
 
 	slen = tmpl_afrom_substr(map, &map->rhs,
-				 &FR_SBUFF_IN(value, strlen(value)),
+				 &FR_SBUFF_IN(value, slen),
 				 type,
-				 value_parse_rules_unquoted[type],	/* We're not searching for quotes */
+				 p_rules,
 				 rhs_rules);
```

Comment:
`unescaped_value` should be freed after the call to `tmpl_afrom_substr()`.  The value isn't used any more, and there's no reason to keep it around.

Reply:
Done